### PR TITLE
Closure killer for IterateNodeSegments

### PR DESCRIPTION
**Pull request by DaEgi01**
Continuation of the ClosureKiller project.
This time it hit IterateNodeSegments to be replaced with GetNodeSegmentIds.
Check out each commit and its full description to really understand what is going on :)
In the last commit (ignoring the master merge, that was just to fix potential conflicts and ignoring the dll path fix) I provided a struct based iterator for zero heap allocations while foreaching over GetNodeSegmentIds.
I guess this could be used as a base for more such stuff, just keep in mind that its ugly, verbose and has potential pitfalls since its a mutable struct and mutable structs are evil :D. Just compare it to the previous yield based operator to see how much uglier it is. So I would advice to use such stuff only in perf critical parts (everything that is not triggered by the user).

### CODE REVIEW

**Comment by DaEgi01** [comment]
Ok, paths are fine now and it builds :)

**Review by krzychu124 — APPROVED** [review]
Good, good

File: TLM/TLM/UI/TrafficManagerTool.cs [hunk 1]
```diff
@@ -615,13 +615,15 @@ void DefaultRenderOverlay(RenderManager.CameraInfo cameraInfo)
         private static float CalculateNodeRadius(ushort nodeId) {
             float sumHalfWidth = 0;
             int count = 0;
-            Constants.ServiceFactory.NetService.IterateNodeSegments(
-                nodeId,
-                (ushort segmentId, ref NetSegment segment) => {
-                    sumHalfWidth += segment.Info.m_halfWidth;
+            ref NetNode node = ref nodeId.ToNode();
+            for (int i = 0; i < 8; ++i) {
```
Comment (kvakvs) [inline TLM/TLM/UI/TrafficManagerTool.cs:619]: For all use of 8 there was a const somewhere in ColossalOrder code.

Comment (krzychu124) [reply TLM/TLM/UI/TrafficManagerTool.cs:619]: I haven't seen such `const` value in the decompiled code 😕

**Review by kvakvs — APPROVED** [review]
Nice

File: TLM/TLM/UI/TrafficManagerTool.cs (hunk 1 above)
Comment (kvakvs) [reply TLM/TLM/UI/TrafficManagerTool.cs:619]: Ok looks like indeed, there isn't. There are other good constants though in NetManager code, but not relevant to this review.

**Pull request merged**